url27a7ae3508_serkan_ozal_ocean_of_memories_tgz-pJ8-com_zeroturnaround_rebellabs_oceanofmemories_article1_objectlayout_ObjectMemoryLayoutDemoJ8/src/com/zeroturnaround/rebellabs/oceanofmemories/common/util/ObjectTree.java

```diff
@@ -0,0 +1,177 @@
+/**
+ * @author SERKAN OZAL
+ *         
+ *         E-Mail: <a href="mailto:serkanozal86@hotmail.com">serkanozal86@hotmail.com</a>
+ *         GitHub: <a>https://github.com/serkan-ozal</a>
+ */
+
+package com.zeroturnaround.rebellabs.oceanofmemories.common.util;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Array;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.IdentityHashMap;
+import java.util.Iterator;
+import java.util.List;
+
+public class ObjectTree {
+
+    public static void dump(PrintWriter pw, Object root) {
+        Node nodeTree = Node.create(root);
+        printTree(new StringBuilder(), new StringBuilder(), pw, nodeTree);
+    }
+
+    public static String dump(Object root) {
+        StringWriter sw = new StringWriter();
+        PrintWriter pw = new PrintWriter(sw);
+        dump(pw, root);
+        pw.flush();
+        return sw.toString();
+    }
+
+    private static class Node {
+
+        private String name;
+        private List<Node> children;
+
+        private long shallowSize;
+        private long deepSize;
+
+        public Node(String name, Object delegate) {
+            this.name = name;
+
+            if (delegate != null) {
+                shallowSize = JvmUtil.shallowSizeOf(delegate);
+                deepSize = shallowSize;
+            }
+        }
+
+        private void addChild(Node node) {
+            if (children == null) {
+                children = new ArrayList<Node>();
+            }
+            children.add(node);
+            deepSize += node.deepSize;
+        }
+
+        public static Node create(Object delegate) {
+            return create("root", delegate,
+                    new IdentityHashMap<Object, Integer>());
+        }
+
+        public static Node create(String prefix, Object delegate,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.